backend/cmm_helpers.ml

```diff
@@ -1045,14 +1045,41 @@ let rec low_32 dbg = function
    (if the word size is 32, this is a no-op) *)
 let sign_extend_32 dbg e =
   if size_int = 4 then e else
-    Cop(Casr, [Cop(Clsl, [low_32 dbg e; Cconst_int(32, dbg)], dbg);
-               Cconst_int(32, dbg)], dbg)
+  match low_32 dbg e with
+  | Cop(Cload ((Thirtytwo_unsigned|Thirtytwo_signed), mut), args, dbg) ->
+     Cop(Cload (Thirtytwo_signed, mut), args, dbg)
+  | e ->
+     Cop(Casr, [Cop(Clsl, [low_32 dbg e; Cconst_int(32, dbg)], dbg);
```

Comment:
can you use `e` instead of `low_32 dbg e` here?

Reply:
Oof, well spotted, thanks.